dotnet/src/SemanticKernel.Core/Functions/KernelFunctionFromPrompt.cs

```diff
@@ -41,13 +42,20 @@ internal sealed class KernelFunctionFromPrompt : KernelFunction
         PromptExecutionSettings? executionSettings = null,
         string? functionName = null,
         string? description = null,
+        string? templateFormat = null,
         IPromptTemplateFactory? promptTemplateFactory = null,
         ILoggerFactory? loggerFactory = null)
     {
         Verify.NotNullOrWhiteSpace(promptTemplate);
 
+        if (promptTemplateFactory is not null)
+        {
+            Verify.NotNullOrWhiteSpace(templateFormat);
```

Comment:
It may be confusing that we are throwing `ArgumentNullException/ArgumentException` for optional parameter, and explanation in XML docs may not be enough. 

One way how to improve it is to provide detailed error message: `Template format is required when prompt template factory is provided`. 

Another way would be to create separate model, where `TemplateFormat` and `PromptTemplateFactory` properties are non-nullable, but actual model is optional in parameters. So, when users want to provide `PromptTemplateFactory`, they will be forced to provide `TemplateFormat` as well (and vice versa, if this scenario is applicable). This solution should reduce the number of error scenarios.

```csharp
public sealed class PromptTemplateParameterers 
{
    public string TemplateFormat { get; set; }
    
    public IPromptTemplateFactory PromptTemplateFactory { get; set; }

    // Constructor...
}
```

```csharp
public static KernelFunction Create(
        string promptTemplate,
        PromptExecutionSettings? executionSettings = null,
        string? functionName = null,
        string? description = null,
        PromptTemplateParameterers promptTemplateParams = null,
        ILoggerFactory? loggerFactory = null)
{
    // Implementation
}
```

Reply:
I've added a better exception message